ui/guest/src/components/Guest.js

```diff
@@ -1142,6 +1166,33 @@ export function Guest(props) {
     }
   }
 
+  function pulseAnimation($element) {
+    $element.removeClass(classes.pulseAnim);
+    setTimeout(function () {
+      $element.addClass(classes.pulseAnim);
+    }, 0);
+  }
+
+  function scrollToElement(node) {
+    let $element;
+    if(node.index !== undefined) {
+      $element = $(`[data-craftercms-model-id="${node.parentId || node.modelId}"][data-craftercms-field-id*="${node.fieldId}"][data-craftercms-index="${node.index}"]`);
+    } else {
+      $element = $(`[data-craftercms-model-id="${node.modelId}"][data-craftercms-field-id="${node.fieldId}"]`);
+    }
+    if($element.length) {
+      if(!isElementInView($element)) {
+        $(scrollElement).animate({
+          scrollTop: $element.offset().top - 100
+        }, 300, function () {
+          pulseAnimation($element);
+        });
+      } else {
+        pulseAnimation($element);
+      }
+    }
+  }
```

Comment:
Looks like all these functions can be outside of the component. Move out all that can be moved out - including others below and above not part of this change set.